testing/matrix/linter/rules/rules.go

```diff
@@ -566,6 +568,53 @@ func objectSecurityContext(object storage.StoreObject) errors.LintRuleError {
 	return errors.EmptyRuleError
 }
 
+func skipServiceTargetPort(o *storage.StoreObject, port int32) bool {
+	// Deployment istiod is defined by its developers so there is no way to specify a named Target Port.
+	if o.Unstructured.GetName() == "istiod" && o.Unstructured.GetNamespace() == "d8-istio" &&
+		(port == 15010 || port == 15012 || port == 15014 || port == 15017) {
+		return true
+	}
+	return false
+}
+
+func objectServiceTargetPort(object storage.StoreObject) errors.LintRuleError {
+	switch object.Unstructured.GetKind() {
+	case "Service":
+	default:
+		return errors.EmptyRuleError
+	}
+
+	converter := runtime.DefaultUnstructuredConverter
+	service := new(v1.Service)
+	err := converter.FromUnstructured(object.Unstructured.UnstructuredContent(), service)
+	if err != nil {
+		panic(err)
+	}
+
+	for _, port := range service.Spec.Ports {
+		if port.TargetPort.Type == intstr.Int {
+			if port.TargetPort.IntVal == 0 {
+				return errors.NewLintRuleError(
+					"MANIFEST004",
+					object.Identity(),
+					nil,
+					"Service port must use an explicit named (non-numeric) target port",
+				)
+			}
+			if skipServiceTargetPort(&object, port.TargetPort.IntVal) {
+				continue
+			}
+			return errors.NewLintRuleError(
+				"MANIFEST004",
```

Comment:
@nabokihms Do we need use different MANIFEST numbers for different, but close checks?

Reply:
No